tests/e2e/steps_light_client_attack.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+// Steps that make carol double sign on the provider, and bob double sign on a single consumer
+func stepsLightClientAttackOnProviderAndConsumer(consumerName string) []Step {
+	return []Step{
+		{
+			// provider double sign
+			action: lightClientEquivocationAttackAction{
+				chain:     chainID("provi"),
+				validator: validatorID("carol"),
+			},
+			state: State{
+				// slash on provider
+				chainID("provi"): ChainState{
+					ValPowers: &map[validatorID]uint{
+						validatorID("alice"): 509,
+						validatorID("bob"):   500,
+						validatorID("carol"): 0, // from 500 to 0
+					},
+				},
+				chainID(consumerName): ChainState{
+					ValPowers: &map[validatorID]uint{
+						validatorID("alice"): 509,
+						validatorID("bob"):   500,
+						validatorID("carol"): 495, // not tombstoned on consumerName yet
+					},
+				},
+			},
+		},
+		{
+			// relay power change to consumerName
+			action: relayPacketsAction{
+				chainA:  chainID("provi"),
+				chainB:  chainID(consumerName),
+				port:    "provider",
+				channel: 0, // consumerName channel
+			},
+			state: State{
+				chainID("provi"): ChainState{
+					ValPowers: &map[validatorID]uint{
+						validatorID("alice"): 509,
+						validatorID("bob"):   500,
+						validatorID("carol"): 0,
+					},
+				},
+				chainID(consumerName): ChainState{
+					ValPowers: &map[validatorID]uint{
+						validatorID("alice"): 509,
+						validatorID("bob"):   500,
+						validatorID("carol"): 0, // tombstoning visible on consumerName
+					},
+				},
+			},
+		},
+		{
+			// consumer double sign
+			// provider will only log the double sign slash
```

Comment:
I do not fully understand the following steps. But by reading the comments I would have expected that we have a consumer double sign, then some packet relaying from consumer to provider that would lead to some power change on the provider. In the following steps we have no power change on neither the provider, nor the consumer. Is this to be expected?

Reply:
There will be no power changes until the `relay` step happens.

Notice that double signs and ligh client attacks are followed by a relay step. No action can be performed on the provider until it receives the correct packet